### PR TITLE
Release for 1.0.0-rc1

### PR DESCRIPTION
**Pull request by lalitb**
Release 1.0.0-rc1

Changes
First candidate release: 1.0.0-rc1

Milestone:  https://github.com/open-telemetry/opentelemetry-cpp/milestone/8
```
1.0.0-rc.1     100% complete
```
For significant contributions please make sure you have completed the following items:

* [x] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

**Review by maxgolov — CHANGES_REQUESTED** [review]
I'd like to get  https://github.com/open-telemetry/opentelemetry-cpp/pull/771 included. Without that change you will not get the promised ABI compatibility on Windows across different Visual Studio versions.

**Review by reyang — APPROVED** [review]
LGTM.

**Comment by reyang** [comment]
> I'd like to get #771 included. Without that change you will not get the promised ABI compatibility on Windows across different Visual Studio versions.

Probably not a blocker for RC?

**Review by ThomsonTan — APPROVED** [review]

**Comment by maxgolov** [comment]
> Probably not a blocker for RC?

I believe it's a blocker. We are committed to be able to provide one build that is compatible with various runtimes.

This is described in our agreed upon policy [here](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/docs/abi-policy.md).

We already have customers using the SDK / API with Visual Studio 2015. Example:
https://github.com/open-telemetry/opentelemetry-cpp/issues/808

We cannot provide any customer the ABI guarantee on Windows until we merge the `absl::variant` PR as the default build option. Also the API surface without this PR is not accepting `const char *` parameters from initializer list, which I believe a major adoption blocker.

File: CHANGELOG.md
Comment (maxgolov) [inline CHANGELOG.md:18]:
```suggestion
## [0.8.0] 2021-06-04
```

**Comment by lalitb** [comment]
> We cannot provide any customer the ABI guarantee on Windows until we merge the `absl::variant` PR as the default build option. Also the API surface without this PR is not accepting `const char *` parameters from initializer list, which I believe a major adoption blocker.

@maxgolov  -  We are feature complete for most of the otel components, have gone through sufficient cycles of alpha/beta release before moving to release candidate. And we don't guarantee ABI compatibility in the release candidate across all the platforms, it's a more complete release but things can still break.  Anyway, will include #771 in rc1 once we can get approvals from others ( I have already reviewed it, looks good, and given my approval yesterday ). Will hold this release for now.

**Review by maxgolov — APPROVED** [review]
LGTM

**Pull request merged**
